### DISCUSSION TITLE
Preference to switch media keys with function keys

### DISCUSSION THREAD

**Discussion by Jopp-gh** [linuxmint/discussions#362] [Pull requests]
I'm asking for a new preference in Mint's keyboard settings in "system settings". 

Basically, to switch from media keys (brightness controls, music controls, mute, etc) to function keys (f1-f12 and f13-f15), globally. 

On my keyboard, very awkwardly, media keys are set as default keys and normal function keys set as secondary keys. I don't get why presets are more important than simple keys. 
Btw, I'm using a Satechi, ‎ST-AMWKM.

Some bios interfaces allow to switch media keys with  function keys, my bios doesn't have such options, unfortunately. I triple checked. 

Most Keyboards have a build-in shortcut (usually fn+esc) to switch function and media keys. The problem here's, you've to repeat this shortcut after every login, this is very cumbersome and very annoying, because I need these function keys all times.

Also, I tried to follow a tutorial for Ubuntu and for my apple-like keyboard :
https://help.ubuntu.com/community/AppleKeyboard#Change_Function_Key_behavior

It didn't work, because users cannot write to system folders. All it takes seems to be a file called "pb_fnmode" in /sys/module/hid/parameters/ with a value of "2"

Why 2?
Here a description of each behavior :

0 = disabled : Disable the 'fn' key. Pressing 'fn'+'F8' will behave like you only press 'F8'
1 = fkeyslast : Function keys are used as last key. Pressing 'F8' key will act as a special key. Pressing 'fn'+'F8' will behave like a F8.
2 = fkeysfirst : Function keys are used as first key. Pressing 'F8' key will behave like a F8. Pressing 'fn'+'F8' will act as special key (play/pause).

**Comment by rcalixte** [comment]
Modifying `/sys/module/hid/parameters/` would be temporary and might be easier to change with a root user cron job. You can use the `@reboot` field for the job.

If you want this to be a permanent one-time change, have you tried the other steps outlined in that link?

> 1. Run the following command to append the configuration line to the file /etc/modprobe.d/hid_apple.conf creating it if necessary:
> ```bash
> $ echo options hid_apple fnmode=2 | sudo tee -a /etc/modprobe.d/hid_apple.conf
> ```
> 2. Trigger copying the configuration into the initramfs bootfile.
> ```bash
> $ sudo update-initramfs -u -k all
> ```

**Reply by Jopp-gh** [reply]
Thanks a lot for your answer rcalixte, my head was already filled with too much tutorials and so I missed the point,  then

I followed the aforementioned ubuntu tutorial,  compared with your suggestions aswell, but couldn't get fn key lock working. 

Probably my Satechi keyboard is not seen as mac alike-keyboard and therefore any related configurations aren't read in either. I asked Satechi's help desk too and ofc, their well meant answers are just pre-written templates, rather than technical advice. 

Switching the fn lock key on isn't the issue, but making this setting sticky,  is. 

Since keyboard / periferial configurations are rather obscure settings they  really should be included by default :(
